UnityProject/Assets/Scripts/Objects/Drawers/Cremator.cs

```diff
@@ -107,55 +121,65 @@ public override void CloseDrawer()
 			UpdateCloseState();
 		}
 
+		public override void OpenDrawer()
+		{
+			base.OpenDrawer();
+			if(drawerState == (DrawerState)CrematorState.ShutAndActive) StopCoroutine(BurnContent());
+		}
+
 		private void UpdateCloseState()
 		{
 			if (container.IsEmpty == false)
 			{
 				SetDrawerState((DrawerState)CrematorState.ShutWithContents);
+				return;
 			}
-			else SetDrawerState(DrawerState.Shut);
+			SetDrawerState(DrawerState.Shut);
 		}
 
 		private void Cremate()
 		{
-			OnStartPlayerCremation();
-			StartCoroutine(PlayIncineratingAnim());
 			SoundManager.PlayNetworkedAtPos(CremationSound, DrawerWorldPosition, sourceObj: gameObject);
+			SetDrawerState((DrawerState)CrematorState.ShutAndActive);
+			UpdateCloseState();
+			OnStartPlayerCremation();
+			StartCoroutine(nameof(BurnContent));
```

Comment:
There isn't anything to stop the burning once complete as it is recursive. Store a reference to this coroutine and stop it on some condition, like timer ending or OnDisable.

Though, does it need to be recursive? I think a loop would be fine here, and don't need to worry about stopping it.

25 damage per five seconds also sounds pretty low.

Reply:
Added an extra check to stop it.

Also 25 per five seconds is not low at all.